packages/shared/lib/contexts/governance/utils/getProposalFromEventId.ts

```diff
@@ -0,0 +1,9 @@
+import { getVotingEvents } from '@core/profile-manager/api'
+import { IProposal } from '../interfaces'
+import { createProposalFromEvent } from './createProposalsFromEvents'
+
+export async function getProposalFromEventId(eventId: string): Promise<IProposal> {
+    const events = await getVotingEvents()
+    const event = events.find((event) => event.id === eventId)
+    return createProposalFromEvent(event)
```

Comment:
This works for now, but we want to be careful using a function like this as the return value. If `createProposalFromEvent` starts doing anything stateful like adding to a store or persisting a value, then everytime the user would use a deep link with an already registered event, it would become duplicated.

Reply:
I will add a comment about it